middle_end/flambda/inlining/inlining_decision.ml

```diff
@@ -148,20 +137,27 @@ let make_decision_for_function_declaration denv ~cost_metrics_source function_de
      first examining call sites. *)
   let code_id = Function_declaration.code_id function_decl in
   let code = DE.find_code denv code_id in
+  let args =
+    Code.inlining_arguments code
+    |> Inlining_arguments.meet (DE.inlining_arguments denv)
+  in
   match Code.inline code with
   | Never_inline -> Never_inline_attribute
   | Hint_inline | Always_inline -> Attribute_inline
   | Default_inline | Unroll _ ->
     if Code.stub code then Stub
     else
-      let round = DE.round denv in
       let metrics =
         match cost_metrics_source with
         | Metrics metrics -> metrics
         | From_denv -> Code.cost_metrics code
       in
-      let large_function_size = get_large_function_size ~round in
-      let small_function_size = get_small_function_size ~round in
+      let large_function_size =
+        Inlining_arguments.large_function_size args |> Code_size.of_int
```

Comment:
Maybe `Inlining_arguments` could directly return a `Code_size.t` for these?

Reply:
I would like it to be possible but in that case [Code_size] would need not to depend on [Inlining_arguments]. It is currently the case but as [Removed_operations] depends on [Inlining_arguments] I'm afraid that this might be a bit confusing and might introduce a circular dependency later on.